subsys/bluetooth/host/keys.h

```diff
@@ -74,19 +77,108 @@ struct bt_keys {
 #define BT_KEYS_STORAGE_LEN     (sizeof(struct bt_keys) - \
 				 offsetof(struct bt_keys, storage_start))
 
-void bt_keys_foreach(int type, void (*func)(struct bt_keys *keys, void *data),
+/**
+ * @brief Get a call through the callback for each key with the same type
```

Comment:
Note that, in our case, it will not be generated even if we set `INTERNAL_DOCS` to `YES` because we only provide `include/` as input for Doxygen, but it's better to be ready in case this changes.

Reply:
@gmarull can you please confirm?

Reply:
The way to _hide_ content from Doxygen:

```c
/** @cond INTERNAL_HIDDEN */

/* whatever here */

/** @endcond */
```

Reply:
But we don't really want to hide it This is not in `include/` so Doxygen will not process it. What we want is to mark it as an internal API (documented with Doxygen) so that if, in the future, we decide to start compiling internal APIs with Doxygen, it'll be ready @gmarull.

Reply:
So using `@internal`/`@endinternal` allows making certain fragments internal, but afaik it will not hide the entity being documented.

Reply:
@ahmedmoheb-nordic why did you set `@code INTERNAL_DOCS`? That´s not what we proposed is it?

Reply:
> @ahmedmoheb-nordic why did you set `@code INTERNAL_DOCS`? That´s not what we proposed is it?

Sorry, I got confused.

So, I should wrap the comment with these `@internal/@endinternal`, correct?

Reply:
@carlescufi I searched for similar uses and I found that only `@internal` needs to added to the comment section that you don't want to handle, please confirm?

Reply:
My understanding here is that you want to exclude a set of APIs from Doxygen, am I right? Then just use https://github.com/zephyrproject-rtos/zephyr/pull/49596/files#r961716485. You can put multiple functions inside a cond/endcond block.

Reply:
> @ahmedmoheb-nordic why did you set `@code INTERNAL_DOCS`? That´s not what we proposed is it?

I've replaced it with `@internal` which I think it has the same effect as wrapping the comment section with `/** @cond INTERNAL_DOCS */` `/** @endcond */` except it's simpler.

Reply:
Turns out this doesn't work @ahmedmoheb-nordic, sorry about that, my mistake. Can you please instead use `/** @cond INTERNAL_HIDDEN */` and `/** @endcond */`, but only once in the file, like here:
https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/nordic_nrf/common/pinctrl_soc.h#L23